src/model.coffee

```diff
@@ -169,6 +169,24 @@ class Model extends Backbone.Model
       )
       .promise(options.returnValues.jqXhr)
 
+  destroy: (options = {}) ->
+    cleanUpAssociatedReferences = =>
+      for collectionName, collection of @storageManager.collections
+        for reference, modelName of collection.modelKlass.associations
+          associationKey = collection.modelKlass.associationDetails(reference).key
+          if _.isArray(modelName) && _.find(modelName, (m) => inflection.singularize(m) == @className())
+            collection.storage.each (model) =>
+              model.set(associationKey, _.without(model.get(associationKey), @id))
+          else if !_.isArray(modelName) && inflection.singularize(modelName) == @className()
+            collection.storage.each (model) =>
```

Comment:
To prevent an extra closure from being created, we can pass `this` as the final argument to `.each`. This would look like so

```
collection.storage.each (model) ->
  ...
, this
```